### PR TITLE
feat: add reject loan component

### PR DESCRIPTION
**Pull request by BLasan**
## Description
Reject Loan component created.

## Related issues and discussion
#830 

## Screenshots, if any
Screenshot from 2020-06-06 18-02-51

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by BLasan** [comment]
@muskankhedia  Please review

File: src/app/loans/loans-view/reject-loan/reject-loan.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   */
+  submit() {
+    const rejectedOnDate = this.rejectLoanForm.value.rejectedOnDate;
+    const dateFormat = 'yyyy-MM-dd';
+    this.rejectLoanForm.patchValue({
+      rejectedOnDate: this.datePipe.transform(rejectedOnDate, dateFormat)
+    });
+    const rejectForm = this.rejectLoanForm.value;
+    rejectForm.locale = 'en';
+    rejectForm.dateFormat = dateFormat;
```
Comment (muskankhedia) [inline src/app/loans/loans-view/reject-loan/reject-loan.component.ts:71]: Please add the notes parameter as well.

Comment (BLasan) [reply src/app/loans/loans-view/reject-loan/reject-loan.component.ts:71]: But it's having a form control name

Comment (muskankhedia) [reply src/app/loans/loans-view/reject-loan/reject-loan.component.ts:71]: Oh Yes. Sorry I missed that.

Comment (BLasan) [reply src/app/loans/loans-view/reject-loan/reject-loan.component.ts:71]: No prob :)

**Comment by BLasan** [comment]
Build failed due to time out

**Comment by Rishabh106** [comment]
@BLasan You should remove "Fixes: {{Issue Number}}" from the PR description as it is only needed in the commit message.

**Comment by BLasan** [comment]
@muskankhedia Please review

**Comment by BLasan** [comment]
Travis Error due to time out

File: src/app/loans/loans-routing.module.ts [hunk 2]
```diff
@@ -111,10 +112,13 @@ const routes: Routes = [
           }
         },
         {
-          path: ':action',
-          component: LoanAccountActionsComponent,
-          data: { title: extract('Loan Account Actions'), routeParamBreadcrumb: 'action' },
-        },
```
Comment (muskankhedia) [inline src/app/loans/loans-routing.module.ts:117]: @BLasan You don't need to remove the routing for ':action', that's a generic routing to be used for all the action buttons. You can try using the close action and look for its implementation, the other actions have to be implemented in the same way. And as the action route is made generic you don't need to add separate routes for every button.

Also there is a convention for the action buttons, and its like, from the name of the button in the web-app, replacing the spaces with '-' and making it lowercase.You can find the code for the same in the file, src/app/loans/loans-view/loans-view.component.ts, the loanAction().

**Comment by BLasan** [comment]
@muskankhedia Please review

File: src/app/loans/loans-view/loan-account-actions/reject-loan/reject-loan.component.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    const rejectedOnDate = this.rejectLoanForm.value.rejectedOnDate;
+    const dateFormat = 'yyyy-MM-dd';
+    this.rejectLoanForm.patchValue({
+      rejectedOnDate: this.datePipe.transform(rejectedOnDate, dateFormat)
+    });
+    const rejectForm = this.rejectLoanForm.value;
+    rejectForm.locale = 'en';
+    rejectForm.dateFormat = dateFormat;
+    this.loanService.rejectLoan(this.loanData.id, rejectForm).subscribe((response: any) => {
+      this.router.navigate(['../'], { relativeTo: this.route });
```
Comment (muskankhedia) [inline src/app/loans/loans-view/loan-account-actions/reject-loan/reject-loan.component.ts:73]:
```suggestion
      this.router.navigate(['../general'], { relativeTo: this.route });
```

**Comment by BLasan** [comment]
@muskankhedia Done. Please review

**Comment by muskankhedia** [comment]
@BLasan In the community app code, there are some more parameters linked with the reject loan action option, I will have a check later and review it.

File: src/app/loans/loans-view/loan-account-actions/reject-loan/reject-loan.component.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  submit() {
+    const rejectedOnDate = this.rejectLoanForm.value.rejectedOnDate;
+    const dateFormat = 'yyyy-MM-dd';
+    this.rejectLoanForm.patchValue({
+      rejectedOnDate: this.datePipe.transform(rejectedOnDate, dateFormat)
+    });
+    const rejectForm = this.rejectLoanForm.value;
+    rejectForm.locale = 'en';
+    rejectForm.dateFormat = dateFormat;
+    this.loanService.rejectLoan(this.loanId, rejectForm).subscribe((response: any) => {
```
Comment (muskankhedia) [inline src/app/loans/loans-view/loan-account-actions/reject-loan/reject-loan.component.ts:70]:
```suggestion
    this.loanService.loanActionButtons(this.loanId, "reject", rejectForm).subscribe((response: any) => {
```

File: src/app/loans/loans.service.ts [hunk 5]
```diff
@@ -42,6 +42,12 @@ export class LoansService {
     return this.http.get(`/loans/${loanId}`);
   }
 
+  rejectLoan(loanId: string, rejectData: any): Observable<any> {
```
Comment (muskankhedia) [inline src/app/loans/loans.service.ts:45]: As now the function has been generalised, this can be removed.

File: src/environments/environment.ts [hunk 6]
```diff
@@ -9,8 +9,8 @@ import env from './.env';
 export const environment = {
   production: false,
   version: env.npm_package_version + '-dev',
-  fineractPlatformTenantId: 'default',  // For connecting to server running elsewhere update the tenant identifier
-  baseApiUrl: 'https://dev.mifos.io',  // For connecting to server running elsewhere update the base API URL
+  fineractPlatformTenantId: 'staging',  // For connecting to server running elsewhere update the tenant identifier
```
Comment (muskankhedia) [inline src/environments/environment.ts:12]: Please revert this

**Comment by muskankhedia** [comment]
@BLasan Also please change the commit message and the PR title to `feat: add loan action (reject)`

**Comment by BLasan** [comment]
> @BLasan Also please change the commit message and the PR title to `feat: add loan action (reject)`

Done

File: src/app/loans/loans-view/loan-account-actions/loan-account-actions.component.html [hunk 7]
```diff
@@ -14,5 +14,7 @@
     <mifosx-view-guarantors [dataObject]="actionButtonData" *ngIf="actions.view_guarantors"></mifosx-view-guarantors>
     <mifosx-create-guarantor [dataObject]="actionButtonData" *ngIf="actions.create_guarantor"></mifosx-create-guarantor>
     <mifosx-disburse-loan-account [dataObject]="actionButtonData" *ngIf="actions.disburse_to_savings"></mifosx-disburse-loan-account>
+    <mifosx-loans-account-close *ngIf="actions.close"></mifosx-loans-account-close>
```
Comment (muskankhedia) [inline src/app/loans/loans-view/loan-account-actions/loan-account-actions.component.html:17]: Please remove this component, it's already declared above.

**Comment by BLasan** [comment]
@muskankhedia  Done

**Pull request merged**